### PR TITLE
VLAN/VE/LIF support

### PR DESCRIPTION
**Pull request by mdurrant-b3**

### CODE REVIEW

**Comment by bosatsu** [comment]
This will be an incredibly helpful feature to have!  Are there any plans to add support for v21 client?

File: acos_client/v30/interface.py [hunk 1]
```diff
@@ -137,11 +140,38 @@ class LogicalInterface(Interface):
...
+            payload = self._build_payload(ifnum=ifnum, ip_address=ip_address, ip_netmask=ip_netmask,
+                                          dhcp=dhcp, enabled=enable, speed=speed,
+                                          default_gateway=default_gateway)
+            return self._post(self.url_prefix,
+                              payload)
     
-    def _build_payload(self, **kwargs):
-        rv = super(LogicalInterface, self)._build_payload(**kwargs)
-        return rv
+    def _build_payload(self, ifnum=None, ip_address=None, ip_netmask=None, dhcp=False, enable=None, enabled=None, speed="auto", default_gateway=None):
```
Comment (jak3kaj) [inline acos_client/v30/interface.py]: Is this a typo?  `enable=None, enabled=None,`

File: acos_client/v30/interface.py [hunk 2]
```diff
@@ -137,11 +140,38 @@ class LogicalInterface(Interface):
     def __init__(self, client):
         super(LogicalInterface, self).__init__(client)
         self.iftype = "lif"
+        self.url_prefix = "{0}{1}/".format(self.url_prefix, self.iftype)
+
+        def create(self, ifnum=None, ip_address=None, ip_netmask=None, dhcp=False, enable=None,
```
Comment (jak3kaj) [inline acos_client/v30/interface.py]: Typo?  `enable=None`

File: acos_client/v30/interface.py [hunk 3]
```diff
@@ -137,11 +140,38 @@ class LogicalInterface(Interface):
...
+                   speed="auto", default_gateway=None):
+            payload = self._build_payload(ifnum=ifnum, ip_address=ip_address, ip_netmask=ip_netmask,
+                                          dhcp=dhcp, enabled=enable, speed=speed,
```
Comment (jak3kaj) [inline acos_client/v30/interface.py]: Typo?   `enabled=enable`
Do you want to use `enable` or `enabled`?

**Comment by mdurrant-b3** [comment]
In the near future, yes. We're currently doinga POC requiring configuration of these elements. Once I get that delivered I'll backport the v21 stuff.

________________________________

File: acos_client/v30/vlan.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                                      tagged_eths, tagged_trunks, 
+                                      veth_interface, logical_interface)
+
+    def get(self, vlan_id):
+        return self._get(self._build_id_url(vlan_id))
+
+    def delete(self, vlan_id):
+        return self._delete(self._build_id_url(vlan_id))
+
+    def _build_payload(self, vlan_id, shared_vlan,
```
[the remark this answers is not in the document]
Comment (mdurrant-b3) [reply acos_client/v30/vlan.py]: This function addresses whether they exist in the schema. Having them as keywords to the main create function with a default of None allow for schema-compliant consumption. For example, shared_vlans isn't added if it's not explicitly enabled and we only add (un)tagged-(trunks|eths) as specified.

It does need to raise an exception if mutual exclusive things like untagged_eths/tagged_eths are passed, but given that it's responsibility is formatting data correctly to send to the API and not enforcing the business rules, I've tried to leave those out as much as possible.

**Pull request merged**